Minio/ApiEndpoints/ObjectOperations.cs

```diff
@@ -690,8 +692,93 @@ private IObservable<Upload> listIncompleteUploads(string bucketName, string pref
+            List<XElement> objects = new List<XElement>();
+
+            for (int i = 0; i < objectsList.Count; i++)
+            {
+                objects.Add(new XElement("Object",
+                                       new XElement("Key", objectsList[i].Key)));
+            }
+
+            var deleteObjectsRequest = new XElement("DeleteObject", objects,
+                                        new XElement("Quiet",true));
+
+            var bodyString = deleteObjectsRequest.ToString();
+
+            var body = System.Text.Encoding.UTF8.GetBytes(bodyString);
+            request.AddParameter("application/xml", body, RestSharp.ParameterType.RequestBody);
+            var response = await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken);
+            var contentBytes = System.Text.Encoding.UTF8.GetBytes(response.Content);
+            DeleteObjectsResult deleteResult = null;
+            using (var stream = new MemoryStream(contentBytes))
+                deleteResult = (DeleteObjectsResult)(new XmlSerializer(typeof(DeleteObjectsResult)).Deserialize(stream));
+
+            if (deleteResult == null )
+            {
+                return new List<DeleteError>();
+            }
+            return deleteResult.ErrorList();
+        }
 
+        /// <summary>
+        /// Removes multiple objects from a specific bucket
+        /// </summary>
+        /// <param name="bucketName">Bucket to  remove objects from</param>
+        /// <param name="objectNames">List of object keys to remove.</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
+        /// <returns></returns>
+        public async Task<IObservable<DeleteError>> RemoveObjectAsync(string bucketName, IEnumerable<string> objectNames, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (objectNames == null)
+            {
+                return null;
+            }
+            utils.validateBucketName(bucketName);
+            List<DeleteObject> objectList;
+            return Observable.Create<DeleteError>(
+              async obs =>
+              {
+                  bool process = true;
+                  int count = objectNames.Count();
+                  int i = 0;
 
+                  while (process)
+                  {
+                      objectList = new List<DeleteObject>();
+                      while (i < count)
+                      {
+                          string objectName = objectNames.ElementAt(i);
+                          utils.validateObjectName(objectName);
+                          objectList.Add(new DeleteObject(objectName));
+                          i++;
+                          if (i % 1000 == 0)
+                             break;                          
+                      }
+                      if (objectList.Count() > 0)
```

Comment:
we don't need to check if `objectList.Count() > 0`. `errorsList.Count` would be zero if there were no objects in `objectList`.

Reply:
s3 multi object delete operation expects at least one <Object> entry for the call. Hence the check.